### ISSUE TITLE
Nested many fields and `pass_original`

### ISSUE THREAD

**Issue by JBKahn** [marshmallow-code/marshmallow#462]
I took a quick look at the code and I'm not sure it's a trivial change or if you'd be interested but in my application, I want to verify that no unwanted fields are passed. When I have a nested field with `many=True`, then I cannot check each one individually for unwanted data being passe. This is because `original_data` is the list of objects and I'm unsure how I can correctly identify which bit of `original_data` corresponds to the `data`.

What are your thoughts?

**Comment by JBKahn** [comment]
Ohhh, `run_validator` already has index, so all you'd need to do is an:
`if index is None then original_data else original_data[index]` and that could be an option? or I think a reasonable default? that or pass the index.

**Comment by vgavro** [comment]
Solution:
```python
class SomeSchema(ma.Schema):
    @ma.post_load(pass_many=True, pass_original=True)
    def __post_load(self, data, many, original_data):
        if hasattr(self.Meta, 'model'):
            if many:
                assert len(data) == len(original_data)
                for i in range(len(data)):
                    data[i]['_entity'] = original_data[i]  # here is your bind
                return tuple(self.Meta.model(**d) for d in data)
            else:
                data['_entity'] = original_data
                return self.Meta.model(**data)
        return maybe_attr_dict(data)
```

**Comment by rooterkyberian** [comment]
I have also run into this issue, but during `post_dump`. I wanted to modify each dumped object by adding some arbitrary fields on dump based on the original content, but with `post_dump(pass_original=True)` when Schema was used in `Nested(..., many=True)` I got a list as a original with no way of telling which element corresponded to currently dumped. I worked around it by adding `pass_many` and adding some `if`s for handling `many=False` and zipping two lists together.

**Comment by sloria** [comment]
I believe this was addressed in [3.0.0b8](https://marshmallow.readthedocs.io/en/3.0/changelog.html#b8-2018-03-24).

Also, the OP's use case (validating that unknown fields are not passed) has first class support with the `unknown` option (https://marshmallow.readthedocs.io/en/3.0/quickstart.html#handling-unknown-fields).

**Comment by JBKahn** [comment]
Thanks!
